driver/makedepend.ml

```diff
@@ -197,25 +199,37 @@ let print_filename s =
 
 let print_dependencies target_files deps =
   let pos = ref 0 in
-  let print_on_same_line item =
+  let dir_sep = if Sys.os_type = "Unix" then '/'
+                else if !Clflags.force_slash then '/' else '\\' in
+  let print_prefix = function
+    | "" -> Fun.id, 0
+    | s ->
+       let len = String.length s in
+       let s, len = if String.get s (len - 1) = dir_sep then s, len
+                    else s ^ String.make 1 dir_sep, len + 1 in
+       (fun () -> print_string s; pos := !pos + len), len
+  in
+  let print_source_tree, source_tree_len = print_prefix !source_tree
+  and print_build_tree, _ = print_prefix !build_tree in
+  let print_on_same_line print_prefix item =
```

Comment:
This code is more complicated that it needs to be: instead of passing `print_prefix` you could just pass the prefix (`!source_tree` or `!build_tree`) and call `print_prefix` with it.

Reply:
I've rewritten the code, I think it's clearer now. Maybe your comment doesn't apply anymore?